invenio_app_ils/errors.py

```diff
@@ -194,9 +205,47 @@ def __init__(self, patron_pid, current_user_pid, **kwargs):
         )
 
 
+class LoanSelfCheckoutItemUnavailable(IlsException):
+    """A patron cannot self-checkout an item."""
+
+    code = 400
+    description = "This literature is not available for self-checkout. Please contact the library for more information."
+
+    def get_body(self, environ=None, scope=None):
+        """Get the request body."""
+        body = dict(
+            status=self.code,
+            message=self.get_description(environ),
+        )
+
+        if self.supportCode:
+            body["supportCode"] = self.supportCode
+
+        return json.dumps(body)
+
+
+class LoanSelfCheckoutItemInvalidStatus(LoanSelfCheckoutItemUnavailable):
+    """A patron cannot self-checkout an item that cannot circulate."""
+
+    supportCode = "SELF-CHECKOUT-001"
+
+
+class LoanSelfCheckoutDocumentOverbooked(LoanSelfCheckoutItemUnavailable):
+    """A patron cannot self-checkout an item for an overbooked document."""
+
+    supportCode = "SELF-CHECKOUT-002"
```

Comment:
I think we should share these codes with the librarians, since they will be the first point of contact

Reply:
Yes, that was the idea indeed ;)